File: utils/model_options.py
"""模型升級時保留既有推理模式，避免依賴供應商的預設值。"""
from typing import Literal, TypedDict


class ReasoningOptions(TypedDict, total=False):
    reasoning_effort: Literal["none"]


def reasoning_options(model: str) -> ReasoningOptions:
    # Luna / Sol 預設 medium；原 5.4 / nano 預設 none。
    # Chat Completions 的 Luna 工具呼叫也需要 none。
    if model in {"gpt-6-luna", "gpt-6-sol"}:
        return {"reasoning_effort": "none"}
    return {}

File: api/src/utils/model-options.ts
/** 保留原 5.4 / nano 的 none 行為，並滿足 Luna 的 Chat Completions 工具呼叫限制。 */
export function reasoningOptions(model: string): { reasoning_effort?: "none" } {
  return model === "gpt-6-luna" || model === "gpt-6-sol"
    ? { reasoning_effort: "none" }
    : {};
}
